### PR TITLE
Serve OpenAPI spec 

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR serves the Open API specification file, which lives under `src/main/resources/static/openapi/`, via the standard HMPPS swagger endpoint: `/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config`.

It also updates the spec itself to include an `ErrorResponse` object where appropriate.

### CODE REVIEW

File: src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml [hunk 1]
```diff
@@ -57,6 +59,10 @@ paths:
       responses:
         '201':
           $ref: '#/components/responses/Goal'
+        '400':
+          $ref: '#/components/responses/400ErrorResponse'
+        '404':
```
Comment (nathanrussell-moj-digital) [inline src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:64]: Minor, and perhaps I'm missing something, but why would POSTing to create a goal result in a 404?
Especially when we say in the description "the latter will be created if it doesn't already exist" ?

**Review by nathanrussell-moj-digital — APPROVED** [review]
Love it; ATOB 👍 😁

File: src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml (hunk 1 above)
Comment (mattwills-moj-digital) [reply src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:64]: In case the `{prisonNumber}` in the path isn't known 👍

Comment (nathanrussell-moj-digital) [reply src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:64]: Hmmmmm ..... is that a thing? .... how might we know if the prison number is valid or not? .... 🤔  ... is that the responsibility of our service? .... 🤔

Comment (mattwills-moj-digital) [reply src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:64]: I haven't fully thought this through, as I'm not familiar with how things work here, but I was thinking our service might want to check the prisonNumber exists via NOMIS, or some other system.

Or to put it another way, what happened in our previous project when an attempt was made, for example, to create a PaperApplication for an ERO that doesn't exist?

**Pull request merged**